sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientTestsLive.cs

```diff
@@ -0,0 +1,274 @@
+        }
+
+        [Test]
+        public async Task AnalyzeSentimentTest()
+        {
+            TextAnalyticsClient client = GetClient();
+            string input = "That was the best day of my life!";
+
+            AnalyzeSentimentResult result = await client.AnalyzeSentimentAsync(input);
+            TextSentiment sentiment = result.DocumentSentiment;
+
+            Assert.AreEqual("Positive", sentiment.SentimentClass.ToString());
+            Assert.IsNotNull(sentiment.PositiveScore);
+            Assert.IsNotNull(sentiment.NeutralScore);
+            Assert.IsNotNull(sentiment.NegativeScore);
+            Assert.IsNotNull(sentiment.Offset);
+            Assert.Greater(1, sentiment.Length);
+        }
+
+        [Test]
+        public async Task AnalyzeSentimentWithLanguageTest()
+        {
+            TextAnalyticsClient client = GetClient();
+            string input = "El mejor test del mundo!";
+
+            AnalyzeSentimentResult result = await client.AnalyzeSentimentAsync(input, "es");
+            TextSentiment sentiment = result.DocumentSentiment;
+
+            Assert.AreEqual("Positive", sentiment.SentimentClass.ToString());
+        }
+
+        [Test]
+        public async Task ExtractKeyPhrasesTest()
+        {
+            TextAnalyticsClient client = GetClient();
+            string input = "My cat might need to see a veterinarian.";
+
+            ExtractKeyPhrasesResult result = await client.ExtractKeyPhrasesAsync(input);
+            IReadOnlyCollection<string> keyPhrases = result.KeyPhrases;
+
+            Assert.AreEqual(2, keyPhrases.Count);
+            Assert.IsTrue(keyPhrases.Contains("cat"));
+            Assert.IsTrue(keyPhrases.Contains("veterinarian"));
+        }
+
+        [Test]
+        public async Task ExtractKeyPhrasesWithLanguageTest()
+        {
+            TextAnalyticsClient client = GetClient();
+            string input = "Mi perro está en el veterinario";
+
+            ExtractKeyPhrasesResult result = await client.ExtractKeyPhrasesAsync(input, "es");
+            IReadOnlyCollection<string> keyPhrases = result.KeyPhrases;
+
+            Assert.AreEqual(2, keyPhrases.Count);
+        }
+
+        [Test]
+        public async Task RecognizeEntitiesTest()
+        {
+            TextAnalyticsClient client = GetClient();
+            string input = "Microsoft was founded by Bill Gates and Paul Allen.";
+
+            RecognizeEntitiesResult result = await client.RecognizeEntitiesAsync(input);
+            IReadOnlyCollection<NamedEntity> entities = result.NamedEntities;
+
+            Assert.AreEqual(3, entities.Count);
+
+            var entitiesList = new List<string> { "Bill Gates", "Microsoft", "Paul Allen" };
+            foreach (NamedEntity entity in entities)
+            {
+                Assert.IsTrue(entitiesList.Contains(entity.Text));
+                Assert.IsNotNull(entity.Type);
+                Assert.IsNotNull(entity.Score);
+                Assert.IsNotNull(entity.Offset);
```

Comment:
It would be good to test the values of Type, Offset and Length, as these should always be the same.  Score could change if the service team updates the model, so just null checking this is fine.

Reply:
I'm worried that by doing that we will be testing the service itself.
What would be the value of looking at the value?

Reply:
I kept thinking about it and got Length, we could check that it is different than 0 .
Still not sure/like the other validations